app/models/application_record.rb

```diff
@@ -8,6 +8,17 @@ def as_json(options = {})
     super
   end
 
+  def serializable_hash(options = nil)
+    return super(options) unless has_attribute?(self.class.inheritance_column)
+
+    options = options.try(:dup) || {}
+
+    options[:methods]  = Array(options[:methods]).map(&:to_s)
+    options[:methods] |= Array(self.class.inheritance_column)
+
+    super(options)
```

Comment:
@hsong-rh Can we add a spec for this .We would have to make a REST call to fetch the Task and look at the task type field to see if it is populated.

Reply:
Added